### ISSUE TITLE
Delay WorkManager initialization until after startup

### ISSUE THREAD

**Issue by mcomella** [mozilla-mobile/fenix#11698]
I remember WorkManager being initialized on startup (when looking at profiles). It seems like we can delay initialization until later: https://developer.android.com/topic/libraries/architecture/workmanager/advanced/custom-configuration

But perhaps I misremember. :grin:

**Comment by mcomella** [comment]
It looks like it gets started here: https://searchfox.org/mozilla-mobile/rev/d81a3c3fce58d3a6c3ebd3257ad1566c9bd462ef/FirefoxReality/app/src/common/shared/org/mozilla/vrbrowser/browser/Services.kt#75

I wonder if we can delegate this to a-c/a-s.

**Comment by sraturi** [comment]
PR related to this: https://github.com/MozillaReality/FirefoxReality/pull/3672

**Comment by Amejia481** [comment]
There are some components that rely `WorkManager` on we should do a extensively round of testing on glean/sync/add-ons before landing cc @csadilek @grigoryk

**Comment by sraturi** [comment]
### Result on S7 Fennec Nightly

Changes were having work manager on-demand initialization
https://developer.android.com/topic/libraries/architecture/workmanager/advanced/custom-configuration

Running FNPRMS on Master App link
| No Changes |With Changes  |
|--|--|
|2.9954  |2.96003  |

Running FNPRMS on Master hanoob link

| No Changes |With Changes  |
|--|--|
|1.264769231  |1.253846155  |

**Comment by mcomella** [comment]
@sraturi Those are good results! :) As I understand it, this change causes WorkManager to move into a lazily-loaded state. That means we have to prevent anyone from calling `WorkManager` on startup if we want to gain the benefits. Do you agree?

As a sanity check, with the changes above, does `WorkManager` init appear in startup profiles? I'm wondering if someone is calling `WorkManager` somewhere else by any chance. 

---

If we implement this, I wonder if we should throw an exception from `getWorkManagerConfiguration` if it's called during cold startup.

**Comment by sraturi** [comment]
@mcomella 

> we have to prevent anyone from calling WorkManager on startup if we want to gain the benefits. Do you agree?

That seems right. we seem to waste around 20 ms initializing the work manager.

> does WorkManager init appear in startup profiles? I'm wondering if someone is calling WorkManager somewhere else by any chance.

For the on-demand init profile, I didn't see the work manager getting init anywhere. Also, I tried to init the work manager on the home fragment. It initialized fine (meaning it wasn't used yet?). Then I tried to init it again and got a crash (expected for initializing twice). The first initialization took around 20 milliseconds.
